qutip/core/cy/coefficient.pyx

```diff
@@ -12,23 +12,23 @@ cimport cython
 import qutip
 
 cdef extern from "<complex>" namespace "std" nogil:
-    double complex cos(double complex x)
     double complex conj(double complex x)
     double         norm(double complex x)
-    double complex exp(double complex x)
+
 
 cdef class Coefficient:
     cdef double complex _call(self, double t) except *:
         return 0j
 
-    cpdef void arguments(self, dict args) except *:
-        self.args.update(args)
+    def replace(self, *, arguments=None, tlist=None):
+        """Return the Coefficient with args or tlist changed. """
+        # Cases where tlist or args are supported are managed in those classes
+        return self
 
     def __call__(self, double t, dict args={}):
-        """update args and return the
-        """
+        """Update args and return the coefficient value at `t`. """
```

Comment:
"Update" might be a bit misleading now, because it makes it seem like the coefficient is mutable.